src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
+    {
+        $this->trainingData =  $data;
+    }
+
+    /**
+     * prepares the data loaded in order to keep only two dimensional lines and sets keys to numerical
+     * @return $this
+     */
+    public function prepare()
+    {
+        $this->trainingData = array_values(
```

Comment:
I did that a first, but i tought thatit would be useless since someone would eventually want to prepare and train multiple arrays,
I didn't want to force preparation to be done once